# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/markcelemen/flyrank-wk1-run-notebooks/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the **Refresh / Content Opportunity** lane. As an aspiring data scientist, I asked ChatGPT which lanes better suits my field and it said lane 2 because it's the most relevant and most traditional applied data science project of the four.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [7]:
import pandas as pd
import numpy as np

# Ensure the data is loaded
csv_path = './flyrank-wk1-run-notebooks/data/raw/content_refresh_anonymized.csv'
try:
    df = pd.read_csv(csv_path)
except Exception:
    import os
    if 'df' in globals():
        df = globals()['df']
    else:
        raise FileNotFoundError("Could not find the dataset.")

# 1. Total 90-day clicks
total_clicks = df['clicks_90d'].sum()

# 2. Slice of content currently declining
declining_mask = df['trend_direction'] == 'down'
num_declining = declining_mask.sum()
pct_declining = (num_declining / len(df)) * 100
clicks_at_risk = df.loc[declining_mask, 'clicks_90d'].sum()
pct_clicks_at_risk = (clicks_at_risk / total_clicks) * 100

print(f"--- Impact Analysis for Decision / Action ---")
print(f"Total Content Items: {len(df):,}")
print(f"Total 90-Day Clicks across portfolio: {total_clicks:,.0f}")
print(f"Number of Declining Content Items (Need Action): {num_declining:,} ({pct_declining:.1f}% of portfolio)")
print(f"90-Day Clicks currently at risk: {clicks_at_risk:,.0f} ({pct_clicks_at_risk:.1f}% of total traffic)")

--- Impact Analysis for Decision / Action ---
Total Content Items: 30,000
Total 90-Day Clicks across portfolio: 482,920
Number of Declining Content Items (Need Action): 16,262 (54.2% of portfolio)
90-Day Clicks currently at risk: 217,248 (45.0% of total traffic)


* **What decision does this work improve?**
  This work directly improves the **Content Refresh and Optimization Prioritization** decision. Instead of choosing pages to refresh based on intuition, recency, or arbitrary schedules, we identify which specific pages are experiencing true decay but still possess high market demand.

* **Who acts on it?**
  The primary actors are **SEO Managers, Editorial Directors, and Content Writers** who have limited weekly capacity and must allocate their budgets and writing hours where they will yield the highest recovery return.

* **What does a wrong recommendation cost?**
  A wrong recommendation carries two massive costs:
  1. **Direct Editorial Waste (False Positives):** Spending hours rewriting a page that is stable, seasonal, or facing a search intent shift, resulting in zero recovery of traffic.
  2. **Unmitigated Traffic Decay (False Negatives):** Ignoring a high-value page that is actively decaying.
  
  Our portfolio metrics demonstrate the scale of this risk:
  * Across the **30,000** content items, total organic traffic stands at **482,920 clicks** over 90 days.
  * Currently, **16,262 items (54.2% of the portfolio)** are experiencing a downward trend.
  * This decline puts **217,248 clicks (45.0% of all organic traffic)** directly at risk of complete loss if no action is taken.

Optimizing this selection process with data-driven ranking ensures we protect this 45% traffic baseline while minimizing wasted editorial spend.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
# Calculate 2-3 stark metrics illustrating the opportunity scale
import pandas as pd

# Let's find the median age of stale pages and look at the visibility of declining content
stale_pages = df[df['content_age_days'] >= 180]
pct_stale = (len(stale_pages) / len(df)) * 100

# Average impressions of declining pages to show they still have high demand
declining_pages = df[df['trend_direction'] == 'down']
avg_imp_declining = declining_pages['impressions_90d'].mean()

print(f"Stale pages (>= 180 days old): {pct_stale:.1f}% of entire portfolio")
print(f"Average 90-day search impressions on declining pages: {avg_imp_declining:,.0f} views")

Stale pages (>= 180 days old): 60.0% of entire portfolio
Average 90-day search impressions on declining pages: 4,919 views


* **Metric 1 (High Staleness Risk):** **60.0%** of the content portfolio is stale, meaning it has not been refreshed or updated in at least 180 days (6+ months).
* **Metric 2 (High Latent Demand):** Pages currently labeled as declining still command a substantial audience, averaging **4,919 search impressions** over 90 days. This proves that we are prioritizing pages with active search demand, not dead URLs.
* **Metric 3 (Actionable Volume):** Out of 30,000 items, **16,262 items (54.2%)** are actively declining. This provides an excellent data pool to build a highly optimized queue that focuses limited writing resources on high-impression opportunities first.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

* **What I can claim:**
  * **Decision Support:** This model acts strictly as a prioritization tool to rank pages that are statistically likely to decay but still enjoy high search volume.
  * **Observed Correlations:** We can show association (e.g., lower word counts or long intervals since updates are statistically linked to declining trends).
  * **Historical Performance Patterns:** We can validate how well our model flags historical trends using past observation windows.

* **What I cannot and will never claim:**
  * **Causal Proof:** We cannot claim that updating/rewriting a page *caused* its recovery, since this dataset lacks randomized experimental controls.
  * **Predicting the Google Search Algorithm:** This model predicts patterns in our site performance metrics, not Google's internal, proprietary search ranking weights.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.